app/services/moves/exporter.rb

```diff
@@ -0,0 +1,144 @@
+# frozen_string_literal: true
+
+require 'csv'
+
+module Moves
+  class Exporter
+    attr_reader :moves
+
+    HEADINGS = [
+      'Status',
+      'Reference',
+      'Move type',
+      'Created at',
+      'Last updated',
+      'From location name',
+      'From location code',
+      'To location name',
+      'To location code',
+      'Additional information',
+      'Date of travel',
+      'PNC number',
+      'Prison number',
+      'Last name',
+      'First name(s)',
+      'Date of birth',
+      'Gender',
+      'Ethnicity',
+      'Ethnicity code',
+      'Violent',
+      'Violent details',
+      'Escape',
+      'Escape details',
+      'Must be held separately',
+      'Must be held separately details',
+      'Self harm',
+      'Self harm details',
+      'Concealed items',
+      'Concealed items details',
+      'Any other risks',
+      'Any other risks details',
+      'Special diet or allergy',
+      'Special diet or allergy details',
+      'Health issue',
+      'Health issue details',
+      'Medication',
+      'Medication details',
+      'Wheelchair user',
+      'Wheelchair user details',
+      'Pregnant',
+      'Pregnant details',
+      'Any other requirements',
+      'Any other requirements details',
+      'Solicitor or other legal representation',
+      'Solicitor or other legal representation details',
+      'Sign or other language interpreter',
+      'Sign or other language interpreter details',
+      'Any other information',
+      'Any other information details',
+      'Not for release',
+      'Not for release details',
+      'Not to be released',
+      'Not to be released details',
+      'Requires special vehicle',
+      'Requires special vehicle details',
+      'Uploaded documents',
+    ].freeze
+
+    MOVE_INCLUDES = [:from_location, :to_location, :profile, :documents, person: %i[gender ethnicity]].freeze
+
+    def initialize(moves)
+      @moves = moves
+    end
+
+    def call
+      Tempfile.new('export', Rails.root.join('tmp')).tap do |file|
+        csv = CSV.new(file)
+        csv << HEADINGS
+        moves.includes(MOVE_INCLUDES).find_in_batches do |batched_moves|
```

Comment:
Defaults to batches of 1000 - it's doubtful we would ever go higher than this but I've tested successfully with 11k moves and this holds up fine. Without using batches the eager fetch SQL queries and memory usage would be huge.

Reply:
If you do the following:

```suggestion
        moves.includes(MOVE_INCLUDES).find_each do |move|
```

It will only instantiate 1000 moves at a time and yield at the level (move) that you actually want.

Reply:
I'll investigate that suggestion, thanks